wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java

```diff
@@ -111,7 +111,9 @@ public Response callback(UriInfo uriInfo, @Nullable List<String> errorValues)
               EnvironmentContext.getCurrent().getSubject().getUserId(),
               null,
               null,
-              NameGenerator.generate(OAUTH_2_PREFIX, 5),
+              NameGenerator.generate(
+                  // Do not pass oauth2 as a username to Git credentials for Bitbucket
+                  "bitbucket".equals(providerName) ? providerName + "-" : OAUTH_2_PREFIX, 5),
```

Comment:
please, extract the logic to the separate method with a proper javadoc

Reply:
@vinokurig is the providee name the same for on-prem Bitbucket server and SaaS ?

Reply:
yes, it is the same. We have `bitbucket-server` implementation but it handles oauth1 only and does not use this logic. OAuth2 flow from both bitbucket and bitbucket server is handled by the `bitbucket` implementation.